classes/test/CustomClassLoader3.java

```diff
@@ -0,0 +1,37 @@
+package classes.test;
+
+public class CustomClassLoader3 extends ClassLoader {
+
+  public CustomClassLoader3 (){
+    super(CustomClassLoader3.class.getClassLoader());
+  }
+
+  // This class cannot rely on method/class/etc ordering, as it is
+  // non-standardized.
+  public static void main(String [] args) throws Exception{
+    CustomClassLoader3 ccl = new CustomClassLoader3();
+    String[] signers = new String[]{"One", "Two"};
+    ccl.testSigners(java.lang.Byte.TYPE, null, null);
+    ccl.testSigners(java.lang.Byte.TYPE, signers, null);
+    ccl.testSigners(java.lang.Void.TYPE, signers, null);
+    ccl.testSigners(ccl.getClass(), signers, signers);
+    ccl.testSigners(ccl.getClass(), null, null);
```

Comment:
Can you add a case that tries to set signers on an array type? (Make it a reference array type, e.g., `CustomClassLoader3[]`, rather than a primitive array type.) Then we will have covered all of the possible cases.